### PR TITLE
Added support for Retry-After

### PR DESCRIPTION
**Pull request by whiskeysierra**

### CODE REVIEW

**Comment by whiskeysierra** [comment]
I'd expect exceptions to be more expensive than regex.

**Comment by duergner** [comment]
👍

File: riptide-failsafe/src/main/java/org/zalando/riptide/failsafe/RetryAfterDelayFunction.java [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+    private boolean onlyDigits(final String s) {
+        return digit.matcher(s).matches();
+    }
+
+    private long parseSeconds(final String retryAfter) {
+        return parseLong(retryAfter);
+    }
+
+    private Instant parseDate(final String retryAfter) {
+        return Instant.from(RFC_1123_DATE_TIME.parse(retryAfter));
```
Comment (lukasniemeier-zalando) [inline riptide-failsafe/src/main/java/org/zalando/riptide/failsafe/RetryAfterDelayFunction.java]: Any protection against `DateTimeException` on failure to parse necessary?

**Comment by lukasniemeier-zalando** [comment]
:+1:

**Pull request merged**